#39 importing categories (#66)
nickmoreton/wagtail-wordpress-import: docs/categories.md

# Categories

- [Categories](#categories)
  - [Configuration](#configuration)
  - [Example Category model](#example-category-model)
  - [Example Page model](#example-page-model)

The package can handle the import of categories. Categories are imported as a ManyToMany relationship on a Page model.

## Configuration

It is disabled by default but you can enable the plugin by adding the following to your settings file.

```python
WAGTAIL_WORDPRESS_IMPORT_CATEGORY_PLUGIN_ENABLED = True
```

The plugin will also need to know which model in you app is used for the categories so add the following to your settings file.

```python

> WAGTAIL_WORDPRESS_IMPORT_CATEGORY_PLUGIN_MODEL = "pages.models.Category

Reviewer: **Suggestion**: fix typo

```suggestion
WAGTAIL_WORDPRESS_IMPORT_CATEGORY_PLUGIN_MODEL = "pages.models.Category"
```

```

> **Name field minimum length**: We implemented this because in testing we found that many categories of a single character in length. Name's that are only a single character aren't imported.

Reviewer: **Suggestion**: reword

```suggestion
**Name field minimum length**: Some WordPress sites can have categories just a single character in length. If these are not useful, they can be skipped using this setting.
```

You can alter the minimum length to suit your needs by adding the following to your settings file.

```python

> WAGTAIL_WORDPRESS_IMPORT_CATEGORY_PLUGIN_MIN_NAME_LENGTH = [num] as an integer

Reviewer: **Suggestion**: Use syntactically correct example code

```suggestion
WAGTAIL_WORDPRESS_IMPORT_CATEGORY_PLUGIN_MIN_NAME_LENGTH = 3  # for example
```
